### ISSUE TITLE
Memory leak on windows, python 

### ISSUE THREAD

**Issue by jdedecca** [FRESNA/PyPSA#4]
Consecutive PyPSA runs accumulate a dictionary, increasing memory use.
PyPSA_test_leak attached results in the following mem_top(limit = 3) output. The most referenced dict is the memory leak.
Failed solutions:
gc.collect()
Deleting graph = GraphDesc() lines in components.py
Updating packages (pyomo, scipy, pandas, networkx) - see attached installed packages list

Occurs in windows 7 and 10, python 3.4.4

[pip list.txt](https://github.com/FRESNA/PyPSA/files/228155/pip.list.txt)
[PyPSA_test_leak.txt](https://github.com/FRESNA/PyPSA/files/228154/PyPSA_test_leak.txt)

mem_top() output:

refs:
**3562	<class 'dict'> {218600272: [197884176], 212671344: [193626240], 217596272: [197558288], 220658144: [198935936], 189**
3006	<class 'dict'> {497410048: <weakref at 0x00000000020DF0E8; to 'type' at 0x000000001DA5E000 (islice)>, 8791215081472
2590	<class 'dict'> {'ERROR_USER_PROFILE_LOAD': 500, 'MK_E_NOINVERSE': -2147221012, 'E_INVALIDARG': -2147024809, 'ERROR_

**Comment by coroa** [comment]
As communicated on the mailing list, we're unfortunately unable to reproduce it with python 3.4 or python 2.7 on Debian.

Please verify two things for us with the output of this variation to your MWE:

```
from mem_top import mem_top
import gc
import os
from pypsa.test.test_ac_dc_lopf import test_opf

for i in range(20):
    test_opf()

offender = max(gc.get_objects(), key=lambda obj: len(gc.get_referents(obj)))
referrers = gc.get_referrers(offender)
print(repr(offender)[:100])
print("referred to by {} objects: ".format(len(referrers)))
for obj in referrers[:5]:
     if isinstance(obj, dict): obj = obj.keys()
     print(repr(obj)[:200])
```

Thanks,
Jonas

**Comment by jdedecca** [comment]
Hi Jonas, here is the output (full output annexed):

{224417632: [201971232], 223469440: [201482048], 190879744: [171076880], 223274128: [195287264], 209
referred to by 1 objects: 
dict_keys(['gc', '__cached__', 'offender', '__loader__', '__name__', 'os', 'referrers', 'obj', '__spec__', 'i', '__doc__', '__builtins__', 'mem_top', '__file__', 'test_opf', '__package__'])

[pypsa_output.txt](https://github.com/FRESNA/PyPSA/files/234021/pypsa_output.txt)

**Comment by coroa** [comment]
This result is basically saying the only one referring to the big dictionary
is the main program context by the offender variable itself.

It is strange, because just before querying for the largest object there
should not have been any reference left and thus garbage collected!

You could splice in a `gc.collect()` right after the 20 test_opf runs. But you said, you already tried that anyway.

I could not find any dictionary within PyPSA itself, which uses dictionaries from numbers to lists of numbers. So I unfortunately can not tell you what is going on.

Keep us posted if you find anything else.

**Comment by jdedecca** [comment]
Indeed the problem remains and gc.collect() has no effect on the top memory users.
Nonetheless I am able to still use PyPSA by calling the script each time.
I will remain testing it in future versions and still try to find the issue.
Thanks!

**Comment by nworbmot** [comment]
A memory leak was fixed with PyPSA 0.6.0. If this did not fix your problem, please reopen the issue. Thanks! Tom

**Comment by jdedecca** [comment]
Hi Tom,
I have run the test with version 0.6 and found no memory leak, so congrats
on the good work.
I did notice an error with the ac-dc-lopf example and Pyomo 4.4.1 which
I'll post on git.
Best,

João
